core/src/main/scala/com/pingcap/tispark/TiDBRelation.scala

```diff
@@ -37,18 +37,44 @@ case class TiDBRelation(session: TiSession,
   @transient val sqlContext: SQLContext
 ) extends BaseRelation
     with InsertableRelation {
-  val table: TiTableInfo = meta
-    .getTable(tableRef.databaseName, tableRef.tableName)
-    .getOrElse(
-      throw new TiClientInternalException(
-        "Table not exist " + tableRef + " valid databases are: " + meta.getDatabases
-          .map(_.getName)
-          .mkString("[", ",", "]")
-      )
-    )
+  lazy val table: TiTableInfo = getTableOrThrow(tableRef.databaseName, tableRef.tableName)
 
   override lazy val schema: StructType = TiUtil.getSchemaFromTable(table)
 
+  private def getTableOrThrow(database: String, table: String): TiTableInfo =
+    meta.getTable(database, table).getOrElse {
+      val db = meta.getDatabase(database)
+      if (db.isEmpty) {
+        throw new TiClientInternalException(
+          "Database not exist " + database + " valid databases are: " + meta.getDatabases
+            .map(_.getName)
+            .mkString("[", ",", "]")
+        )
+      } else {
+        throw new TiClientInternalException(
+          "Table not exist " + tableRef + " valid tables are: " + meta
+            .getTables(db.get)
+            .map(_.getName)
+            .mkString("[", ",", "]")
+        )
+      }
+    }
+
+  lazy val isTiFlashReplicaAvailable: Boolean = {
+    // Note: INFORMATION_SCHEMA.TIFLASH_REPLICA is not present in TiKV.
+    // select * from information_schema.tiflash_replica where table_id = $id
+    // TABLE_SCHEMA, TABLE_NAME, TABLE_ID, REPLICA_COUNT, LOCATION_LABELS, AVAILABLE, PROGRESS
+    table.getTiflashReplicaInfo != null && table.getTiflashReplicaInfo.isAvailable
+  }
+
+  def getTiFlashReplicaProgress: Double = {
```

Comment:
unused function?

Reply:
Not yet used because I wish to use another PR to provide progress status for users. (Perhaps should be presented in desc table result)

Reply:
move this function to next PR?